src/main/java/tech_tasks/array/find/index/IndexInInterval.java

```diff
@@ -0,0 +1,111 @@
+package tech_tasks.array.find.index;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class IndexInInterval {
+
+    @Test
+    void validRangeAndTargetPresentTest() {
+        int[] array = {1, 2, 3, 4, 5};
+        int a = 1, b = 4, target = 3;
+        assertEquals(2, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void targetAtStartOfRangeTest() {
+        int[] array = {1, 2, 3, 4, 5};
+        int a = 0, b = 3, target = 1;
+        assertEquals(0, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void targetAtEndOfRangeTest() {
+        int[] array = {10, 20, 30, 40};
+        int a = 1, b = 4, target = 40;
+        assertEquals(3, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void targetNotInRangeTest() {
+        int[] array = {1, 2, 3, 4, 5};
+        int a = 1, b = 4, target = 5;
+        assertEquals(-1, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void targetAbsentInArrayTest() {
+        int[] array = {1, 2, 3, 4, 5};
+        int a = 0, b = 5, target = 6;
+        assertEquals(-1, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void negativeNumbersInArrayTest() {
+        int[] array = {-10, -20, -30, -40};
+        int a = 1, b = 4, target = -30;
+        assertEquals(2, findIndexInRange(array, a, b, target));
+    }
+
+    @Test
+    void emptyArrayTest() {
+        int[] array = {};
+        int a = 0, b = 0, target = 1;
+        assertThrows(IllegalArgumentException.class, () -> findIndexInRange(array, a, b, target));
```

Comment:
You need to check not only exception, but a text of the exception as well since you have a lot of them.
Example:
```
Exception thrown = assertThrows(IllegalArgumentException.class, () -> findIndexInRange(array, a, b, target));
        System.out.print("Exception message: " + thrown.getMessage());
        assertTrue(thrown.getMessage().contains("Start index 'a' must be less than end index 'b'"));
```

Reply:
Fixed;

Reply:
Don't see a single test with Exception thrown = assertThrows examples